File: app/src/main/java/com/rachelleignacio/listbucket/domain/interactors/GetListBucketInteractor.java
package com.rachelleignacio.listbucket.domain.interactors;

import com.rachelleignacio.listbucket.domain.interactors.base.Interactor;
import com.rachelleignacio.listbucket.domain.models.List;

/**
 * Created by rachelleignacio on 3/9/17.
 */

public interface GetListBucketInteractor extends Interactor {
    interface Callback {
        void onListsRetrieved(java.util.List<List> lists);
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/domain/interactors/GetListBucketInteractor.kt
package com.rachelleignacio.listbucket.domain.interactors

import com.rachelleignacio.listbucket.domain.interactors.base.Interactor
import com.rachelleignacio.listbucket.domain.models.List

/**
 * Created by rachelleignacio on 8/9/17.
 */
interface GetListBucketInteractor : Interactor {
    interface Callback {
        fun onListsRetrieved(lists: MutableList<List>)
    }
}